### ISSUE TITLE
_tag_data() function incorrectly applies padding

### ISSUE THREAD

**Issue by str4d** [AntonKueltz/rfc7539#3]
Per RFC 7539, no padding should be applied to the AAD or ciphertext if they are already an integral multiple of 16 bytes (which includes a zero-length AAD):
> - Finally, the Poly1305 function is called with the Poly1305 key
      calculated above, and a message constructed as a concatenation of
      the following:
>      *  The AAD
>      *  padding1 -- the padding is up to 15 zero bytes, and it brings
         the total length so far to an integral multiple of 16.  If the
         length of the AAD was already an integral multiple of 16 bytes,
         this field is zero-length.
>      *  The ciphertext
>      *  padding2 -- the padding is up to 15 zero bytes, and it brings
         the total length so far to an integral multiple of 16.  If the
         length of the ciphertext was already an integral multiple of 16
         bytes, this field is zero-length.
>      *  The length of the additional data in octets (as a 64-bit
         little-endian integer).
>      *  The length of the ciphertext in octets (as a 64-bit little-
         endian integer).

[Pseudocode](https://tools.ietf.org/html/rfc7539#section-2.8.1):

```
pad16(x):
    if (len(x) % 16)==0
        then return NULL
        else return copies(0, 16-(len(x)%16))
    end
```

However, the current implementation applies 16 bytes of padding in this case:

https://github.com/AntonKueltz/rfc7539/blob/10e0284fb37219983f9972de4f110cc1e1f76cfc/rfc7539/aead.py#L30-L32

**Comment by AntonKueltz** [comment]
Good catch again, using PKCS7 padding currently. I'll update and release later today.

**Comment by AntonKueltz** [comment]
Fixed in commit e7f924b and available in version 1.2.1 on pypi, tag v1.2.1.
